saskatoon/sitebase/templatetags/format_tree_name.py

```diff
@@ -0,0 +1,58 @@
+from django import template
+from django.template.defaultfilters import stringfilter
+
+
+register = template.Library()
+
+TREE_NAME_ICONS = {
```

Comment:
FYI here is a more human readable fixture for tree types:
https://github.com/LesFruitsDefendus/saskatoon-ng/blob/master/saskatoon/fixtures/harvest-treetype.json
You can load it with `python saskatoon/manage.py loaddata saskatoon/fixtures/harvest-treetype.json`

Reply:
> Hello @mina-andajos, Thanks for this, the screenshots looks good.
> 
> But I'm not sure to understand why it works with these names actually, it does not seem to match the names we have in database models (in this [fixture](https://github.com/LesFruitsDefendus/saskatoon-ng/blob/master/saskatoon/fixtures/saskatoon.json)). For instance, we have a couple of cherries: sweet and sour, but both can use the same emoji. 
> 
> Here's the list of fruit names I've extracted from the fixture
> 
> ```
> "fruit_name": "Abricot",
> "fruit_name": "Amélanche",
> "fruit_name": "Cerise sucrée",
> "fruit_name": "Cerise griotte",
> "fruit_name": "Fraise",
> "fruit_name": "Framboise",
> "fruit_name": "Groseille",
> "fruit_name": "Blackberry",
> "fruit_name": "Noix de Grenoble",
> "fruit_name": "Pêche",
> "fruit_name": "Poire rouge",
> "fruit_name": "Poire jaune",
> "fruit_name": "Poire verte",
> "fruit_name": "Pomme verte",
> "fruit_name": "Pomme rouge",
> "fruit_name": "Pomme jaune",
> "fruit_name": "Pommette",
> "fruit_name": "Purple plum",
> "fruit_name": "Green grape",
> "fruit_name": "Raisin mauve (bleu)",
> "fruit_name": "Rhubarbe",
> "fruit_name": "Sureau",
> "fruit_name": "Figue",
> "fruit_name": "Blackcurrant",
> "fruit_name": "Aronia berry",
> "fruit_name": "Mulberry",
> "fruit_name": "Red grapes",
> "fruit_name": "Yellow plum",
> "fruit_name": "Asian pear",
> "fruit_name": "Green plum",
> ```
> 
> I realize that it's both in french and in english, which is a bit confusing. Sorry about that. But why does it work with "Cherries" while this name is not in the fruit list? @2ynn, maybe you have an idea ?
> 
> Thanks,
> 
> Talk to you later


I apologize about the confusion I made.
I have added some tree types in english to the database just to be able to properly test the functionality.

I got a little confused because of the mix of english and french names so I was wondering too if you will be storing them in english and then translating as @2ynn pointed out.

But I will be adding the names from the fixtures that correspond with fruit emojis 👍.

Also maybe if the name is too long where there isn't any space and it doesn't have a corresponding emoji I can display it cut out, like "very long fruit name" > "very...".

Reply:
> @mina-andajos FYI here is a more human readable fixture for tree types:
> https://github.com/LesFruitsDefendus/saskatoon-ng/blob/master/saskatoon/fixtures/harvest-treetype.json
> You can load it with `python saskatoon/manage.py loaddata saskatoon/fixtures/harvest-treetype.json`  
> As you can see the tree name is not the same as the fruit name
> @tristanlatr are we planning on homogenizing the language in fruit names?

Tree name is what's being currently shown on the pages, that's what I'm using to display the icons.
I can display the fruit icons based on the tree's fruit name instead if that's what you would like to do.

Reply:
Yes I think using the `fruit_name` to match with an icon is probably better.

Reply:
Yes I think using the `fruit_name` to match with an icon is probably better.

@2ynn

Each tree has 1 `fruit_name` that corresponds with it.
The templates use `Treetype.__str__` aka. `tree.name` (and not a `Treetype` object so it can't get passed to the template filter).

Should I display `fruit_name` instead of the tree.name for the templates?

Reply:
Which view exactly? I believe we are actually passing `harvest.trees.all()` to the list view, you should be able to use {% t.fruit_name %} no ?

Reply:
> Which view exactly? I believe we are actually passing `harvest.trees.all()` to the list view, you should be able to use {{ t.fruit_name }} no ?

In `harvest/api.py`, the response of `HarvestViewset.retrieve()` and `HarvestViewset.list()` are different. `harvest.trees.all()` are only used in `retrieve()` and not `list()`.

------------------
I tried passing `{{ t.fruit_name }}` in the list templates like `app/list_views/harvest/data_row.html` but it just returns blank text.

![image](https://user-images.githubusercontent.com/52796958/160878646-fbc16eb0-f1e4-490b-b426-7291eee7fe10.png)
![image](https://user-images.githubusercontent.com/52796958/160879110-8c52fdd1-ed9d-412f-8e2c-3cba329b6727.png)

Reply:
ok so i think the proper way would be to create a TreeSerializer and nest in under HarvestSerializer but as I explained in the discussions it's part of a bigger refactoring spree. So instead, feel free to pass additional data to the list view as well for now if that helps you (i.e. `trees`, same as detail view)

Reply:
nevermind, updating the serializer would be best, how comfortable are you with REST?

Reply:
> nevermind, updating the serializer would be best, how comfortable are you with REST?

```python
class HarvestSerializer(serializers.ModelSerializer):

    # three different ways of getting a multimodel serializer:
    # 1) calling a model method
    ...
    # # 2) get string rather than id from a pk
    ...
    trees = serializers.StringRelatedField(many=True)
    # 3) get the full instance from another serializer class
    property = PropertyInfoSerializer(many=False, read_only=True)
```
For the `HarvestSerializer` it's just fetching `treetype.__str__()`. I believe making a `TreeTypeSerializer and then applying it like `property` would be best?